### PR TITLE
BigWig Integration

### PR DESCRIPTION
**Pull request by EricR86**
This PR creates native BigWig support for Genomedata.

To use a BigWig file through the Genomedata interface, you simply create a genome object with a bigWig file instead of a Genomedata file. e.g. `with Genome("myfile.bigWig") as genome`.

A few important notes:
- Tests (should) cover all documented interfaces. Specifically all the ones through the Genome object
- There are a number of public interfaces that are not documented (for example through Chromosome objects). These were covered to the best of my ability but don't necessarily translate well to the BigWig format
- Summary statistics are available through the BigWig header much like the Genomedata file, however these are lower precision since they are stored as integer types. Calculating these manually was attempted but they were very slow.
- There is an implicit single trackname per Genome object (the filename). This is to keep interfaces consistent between types. For example, returning values that would normally be a list indexed per track will now return a list of a single element.
- Each Chromosome is represented with 1 underlying supercontig. Using 'read' on the supercontig is by consequence very slow. Unless the `read` method of the supercontig is used, the entire dataset is never brought into memory.
- Supercontigs/Chromosomes return a numpy array. This might be a slightly different interface than a `tables.EArray` which was the type used by the Genomedata file type.
- Writing and erasing, while technically possible, was not implemented in this PR
- Addressing bigWigs by URL, also while (I believe) technically possible, was not implemented in this PR

### CODE REVIEW

File: genomedata/_bigwig.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+        For :attr:`Genome.format_version` > 0, this will always be 0.
+        For == 0, this will be the start of the first supercontig.
+
+        """
+        # if self._format_version == 0:
+        #     return min(supercontig.start for supercontig in self)
+        # else:
+        #     return self.attrs.start
+        return self._start
```
Comment (michaelmhoffman) [inline genomedata/_bigwig.py:202]: Please remove the dead code and the comment that doesn't apply

Comment (michaelmhoffman) [inline genomedata/_bigwig.py:202]: Please remove the dead code and the docstring that doesn't apply

File: genomedata/_bigwig.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+        """
+        return self._end
+
+
+class _BigWigSupercontig(Supercontig):
+    """A container for a segment of data in one chromosome."""
+
+    class _Continuous():
+        """An indexable-type for the continuous region.
```
Comment (michaelmhoffman) [inline genomedata/_bigwig.py:224]: "indexable-type"?

File: genomedata/_bigwig.py [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+        This type also should implement the .read() method which returns the
+        whole region as a numpy array """
```
Comment (michaelmhoffman) [inline genomedata/_bigwig.py:226]: Seems a weird place to put this note

File: genomedata/_bigwig.py [hunk 4]
```diff
@@ -0,0 +1,290 @@
+import struct
+
+import pyBigWig
+from numpy import array
+
+from ._chromosome import (Chromosome, CONTINUOUS_DTYPE, _ChromosomeList,
+                          _key_to_tuple, Supercontig)
+
+
+BIG_WIG_SIGNATURE = 0x888FFC26
```
[the remark this answers is not in the document]
Comment (EricR86) [reply genomedata/_bigwig.py:10]: I put the path to to the file from kentutils containing the definition as a comment

File: genomedata/_bigwig.py [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+        # Close all the chromosomes so they know they shouldn't be read
+        self.bw_file.close()
+
+        super().close()
+
+
+class _BigWigChromosome(Chromosome):
+
+    def __init__(self, name, big_wig_file, filepath):
+        self.bw_file = big_wig_file
```
[the remark this answers is not in the document]
Comment (EricR86) [reply genomedata/_bigwig.py:109]: Changed to bw throughout

**Pull request merged**